src/callbacks_step/summary.jl

```diff
@@ -17,211 +16,208 @@ beginning of a simulation and then resets the timer. When the returned callback
-    for cb in callbacks.continuous_callbacks
-      show(io_context, MIME"text/plain"(), cb)
-      println(io, "\n")
-    end
-    for cb in callbacks.discrete_callbacks
-      # Do not show ourselves
-      cb.affect! === summary_callback && continue
+    print_startup_message()
 
-      show(io_context, MIME"text/plain"(), cb)
-      println(io, "\n")
-    end
-  else
-    show(io_context, MIME"text/plain"(), callbacks)
+    io = stdout
+    io_context = IOContext(io,
+                           :compact => false,
+                           :key_width => 30,
+                           :total_width => 100,
+                           :indentation_level => 0)
+
+    semi = integrator.p
+    show(io_context, MIME"text/plain"(), semi)
+    println(io, "\n")
+    mesh, equations, solver, _ = mesh_equations_solver_cache(semi)
+    show(io_context, MIME"text/plain"(), mesh)
+    println(io, "\n")
+    show(io_context, MIME"text/plain"(), equations)
+    println(io, "\n")
+    show(io_context, MIME"text/plain"(), solver)
     println(io, "\n")
-  end
-
-  # time integration
-  setup = Pair{String,Any}[
-           "Start time" => first(integrator.sol.prob.tspan),
-           "Final time" => last(integrator.sol.prob.tspan),
-           "time integrator" => integrator.alg |> typeof |> nameof,
-           "adaptive" => integrator.opts.adaptive,
-          ]
-  if integrator.opts.adaptive
-    push!(setup,
-      "abstol" => integrator.opts.abstol,
-      "reltol" => integrator.opts.reltol,
-      "controller" => integrator.opts.controller,
-    )
-  end
-  summary_box(io, "Time integration", setup)
-  println()
-
-  # technical details
-  setup = Pair{String,Any}[
-           "#threads" => Threads.nthreads(),
-          ]
-  if mpi_isparallel()
-    push!(setup,
-      "#MPI ranks" => mpi_nranks(),
-    )
-  end
-  summary_box(io, "Environment information", setup)
-  println()
-
-  reset_timer!(timer())
-
-  return nothing
-end
 
+    callbacks = integrator.opts.callback
+    if callbacks isa CallbackSet
+        for cb in callbacks.continuous_callbacks
+            show(io_context, MIME"text/plain"(), cb)
+            println(io, "\n")
+        end
+        for cb in callbacks.discrete_callbacks
+            # Do not show ourselves
+            cb.affect! === summary_callback && continue
+
+            show(io_context, MIME"text/plain"(), cb)
+            println(io, "\n")
+        end
+    else
+        show(io_context, MIME"text/plain"(), callbacks)
+        println(io, "\n")
+    end
 
-function (cb::DiscreteCallback{Condition,Affect!})(io::IO=stdout) where {Condition, Affect!<:typeof(summary_callback)}
+    # time integration
+    setup = Pair{String, Any}["Start time" => first(integrator.sol.prob.tspan),
+                              "Final time" => last(integrator.sol.prob.tspan),
+                              "time integrator" => integrator.alg |> typeof |> nameof,
+                              "adaptive" => integrator.opts.adaptive]
```

Comment:
Note that, if this is preferred, we should be able to format this as
```suggestion
    setup = Pair{String, Any}[
        "Start time" => first(integrator.sol.prob.tspan),
        "Final time" => last(integrator.sol.prob.tspan),
        "time integrator" => integrator.alg |> typeof |> nameof,
        "adaptive" => integrator.opts.adaptive
    ]
```
by setting the option `variable_call_indent = ["Pair"]` (see my PR [here](https://github.com/domluna/JuliaFormatter.jl/pull/676)).

Reply:
Thanks for pointing that out. To me, both versions look fine. Thus, all other things being equal, I'd vote for the version that requires fewer modifications to the style set.

Reply:
Yes, it is an array - thus the option does not apply - and no, after manually changing it to the proposed layout it will insist on formatting as it is currently formatted.